app/channel.go

```diff
@@ -2892,6 +2893,7 @@ func (a *App) MarkChannelsAsViewed(c request.CTX, channelIDs []string, userID st
 				c.Logger().Warn("Failed to get membership", mlog.Err(err))
 				continue
 			}
+			prevViewedAt[channelID] = member.LastViewedAt
```

Comment:
I injected the lastviewedat (excluding this view) because it seemed more useful than injecting a boolean field isFirstTime. But if you feel that's better to leak less info, I'd be fine with the boolean.